### PR TITLE
fix(nostr): re-encrypt nostr key on pw change #1925

### LINKED ISSUE #1925: Re-encrypt the nostr key when changing the password
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

When changing the password the nostr key is not encrypted with the new password.

### Screenshots [optional]

_No response_

### Steps To Reproduce

-

### Expected behavior

-

### Alby information

master

### Device information

_No response_

### Additional context

_No response_

### Are you working on this?

None

### PR DESCRIPTION
**Pull request by escapedcat**
### Describe the changes you have made in this PR

See title

### Link this PR to an issue [optional]

Fixes #1925

### Type of change

- `fix`: Bug fix (non-breaking change which fixes an issue)

### How has this been tested?

Manually

### CODE REVIEW

**Review by reneaaron — APPROVED** [review]
tACK

**Comment by bumi** [comment]
that is not enough as there are already broken setups out there that currently can not update the key. 
so these fail: https://github.com/getAlby/lightning-browser-extension/blob/master/src/app/screens/Settings.tsx#L40-L43

but the main problem is here: https://github.com/getAlby/lightning-browser-extension/blob/master/src/app/screens/Settings.tsx#L57

File: src/extension/background-script/nostr/index.ts [hunk 1]
```diff
@@ -16,7 +16,11 @@ class Nostr {
     const password = state.getState().password as string;
     const encryptedKey = state.getState().nostrPrivateKey as string;
     if (encryptedKey) {
-      return decryptData(encryptedKey, password);
+      try {
+        return decryptData(encryptedKey, password);
+      } catch (e) {
+        if (e instanceof Error) console.error("Couldn't get key");
```
Comment (bumi) [inline src/extension/background-script/nostr/index.ts:22]:
```suggestion
        if (e instanceof Error) { 
          console.error("Could not decrypt the Nostr key");
          console.error(e);
        }
```

**Comment by reneaaron** [comment]
tACK (tested again with a malformed private key before the upgrade)

**Pull request merged**